meshnet/register.go

```diff
@@ -58,17 +72,39 @@ func (r *RegisteringChecker) IsRegistered() bool {
 		return false
 	}
 
-	return cfg.MeshDevice != nil && cfg.MeshPrivateKey != ""
+	return isRegistered(cfg)
 }
 
-func (r *RegisteringChecker) register(cfg *config.Config) error {
-	if cfg.MeshDevice != nil &&
-		cfg.MeshPrivateKey != "" &&
-		cfg.MeshDevice.ID != uuid.Nil {
-		return nil
+// Register registers the device in API, even if it was already registered
+func (r *RegisteringChecker) Register() error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	var cfg config.Config
+	if err := r.cm.Load(&cfg); err != nil {
+		return err
+	}
+
+	if err := r.register(&cfg); err != nil {
+		return err
+	}
+
+	if err := r.cm.SaveWith(meshConfig(cfg.MeshDevice, cfg.MeshPrivateKey)); err != nil {
```

Comment:
Save config changes after validation is done - I would move this code after last validation below.

Reply:
I did it this way because existing `IsRegistered` function also saves config and then returns the result whether registration succeeded. And I think this way is better, because the only difference is when the previously stored registration was valid, but the new one is not. And if new registration is somehow invalid, I think we would want to just keep retrying it instead of just using the old one. But this is all theoretical of course, I think there's no actual difference in practice.